enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/KeyValidator.java

```diff
@@ -183,10 +186,71 @@ public static APIKeyValidationInfoDTO validateSubscription(String uuid, String a
         return infoDTO;
     }
 
+    public static APIKeyValidationInfoDTO validateSubscription(String apiUuid, String apiContext,
+                                                               JWTClaimsSet payload) {
```

Comment:
This is an new `validateSubscription` overload method for APIkey (and for other self contained jwts if needed) in addition to the `validateSubscription` method right above.

Reply:
The only difference is the app uuid and the key type are taken from the jwt claims in contrast to getting them from the data stored by consumer key and keymanager